### PR TITLE
Blockrv0.0.2

### PR DESCRIPTION
**Pull request by JohnCoene**
This attempts to upgrade to blockr v0.0.2

@DivadNojnarg can you please take a look? I have two issues:

- In `new_falcon02_block` II can't select multiple variables with `new_select_field` despite `multiple` being set to `TRUE`.
- I can't seem to work out how to upgrade the `new_relabel_block` to v0.0.2 without passing it the `data` or default `columns`.

### CODE REVIEW

**Comment by JohnCoene** [comment]
We can just remove that block.

```r
pkgload::load_all()
library(shiny)
library(blockr)

stack1 <- new_stack(
  random_cdisc_data_block,
  falcon02_block,
  title = "Falcon 02"
)

stack2 <- new_stack(
  random_cdisc_data_block,
  falcon05_block,
  title = "Falcon 05"
)

ui <- fluidPage(
  theme = bslib::bs_theme(5L),
  div(
    class = "row",
    div(
      class = "col-md-6",
      generate_ui(stack1)
    ),
    div(
      class = "col-md-6",
      generate_ui(stack2)
    )
  )
)

server <- function(...){
  generate_server(stack1)
  generate_server(stack2)
}

shinyApp(ui, server)
```

**Comment by DivadNojnarg** [comment]
It is normal that falcon was replaced by cardinal? When I run the example I get:

```r
'make_table_02' is not an exported object from 'namespace:falcon'
```

which points to:

```r
falcon::make_table_02(
      df = data,
      vars = .(columns)
    )
```

I have to replace falcon by cardinal everywhere else to make it work. Besides that I get something:

```r
pkgload::load_all()
library(shiny)
library(blockr)

stack1 <- new_stack(
  new_random_cdisc_data_block,
  new_falcon02_block,
  title = "Falcon 02"
)

stack2 <- new_stack(
  new_random_cdisc_data_block,
  new_falcon05_block,
  title = "Falcon 05"
)

ui <- fluidPage(
  theme = bslib::bs_theme(5L),
  div(
    class = "row",
    div(
      class = "col-md-6",
      generate_ui(stack1)
    ),
    div(
      class = "col-md-6",
      generate_ui(stack2)
    )
  )
)

server <- function(...){
  generate_server(stack1)
  generate_server(stack2)
}

shinyApp(ui, server)
```
Screenshot 2024-07-05 at 10 54 54

**Comment by DivadNojnarg** [comment]
I can also reproduce your issue. Need to check what's wrong. Sorry took me some time to install everything.

**Comment by JohnCoene** [comment]
Ah yes, they did that yesterday, I updated the code, we'll rename the package later.

I guess it's a good thing there was already falcon package on CRAN

**Comment by DivadNojnarg** [comment]
@JohnCoene 

I think what happens is that each time the field is updated, the function `first_col` is evaluated which erases the value provided from the UI. 

You could go for this in the meantime:

```r
new_falcon02_block <- function(
  ...,
  default_selected = character()
){
  all_cols <- function(data) colnames(data)

  fields <- list(
    columns = new_select_field(default_selected, all_cols, multiple = TRUE, title = "Columns")
  )
...
}
```

and call it as:

```r
new_falcon02_block(default_selected = c("STUDYID", "AGE"))
```

**Comment by JohnCoene** [comment]
This seems like a bug more than a feature to me, what do you think? This used to work and should work I think.

We do this because the data upstream is dynamic of course.

```r
# error - crashes the app
ardinal::make_table_02(random.cdisc.data::cadlb, vars = character())

# works
ardinal::make_table_02(random.cdisc.data::cadlb, vars = c("SEX"))
```

`new_falcon02_block(default_selected = c("STUDYID", "AGE"))` this only works if we hardcode the block we create but will not work on the fly, or when it's called with defaults `new_falcon02_block())`

This logic of ensuring a column is selected has to then be shifted to the block expression.

```r
col <- .(columns)

if(length(col) == 0L)
  col <- names(data)[1]
```

EDIT:
Note that the above now also means we display something that doesn't match the UI, e.g.: here it will display a table with one column, a column the user has not selected

Let me know if I am missing something.

**Comment by DivadNojnarg** [comment]
Yes I agree that's not ideal. Since @nbenn is making some changes to the fields value API and if the current workaround can work for few days, I would stop here and check again when blockr is updated.

File: R/label_block.R [hunk 1]
```diff
@@ -1,57 +1,51 @@
...
-    values = character(),
-    ...) {
+  columns = colnames(data)[1L],
+  values = character(),
+  ...
+) {
   sub_fields <- function(data, columns) {
     determine_field <- function(x) {
-      string_field
+      blockr::string_field
```
Comment (DivadNojnarg) [inline R/label_block.R:9]: new_string_field?

**Review by DivadNojnarg — APPROVED** [review]

**Pull request merged**